workspaces/ballerina/ballerina-extension/src/rpc-managers/inline-data-mapper/rpc-manager.ts

```diff
@@ -186,6 +190,146 @@ export class InlineDataMapperRpcManager implements InlineDataMapperAPI {
+                this.mergeTextEdits(allTextEdits, result.value.textEdits);
+            } else {
+                console.error(`>>> Failed mapping ${index + 1}:`, result.reason);
+            }
+        });
+
+        return this.optimizeTextEdits(allTextEdits);
+    }
+
+    private mergeTextEdits(
+        allTextEdits: { [key: string]: TextEdit[] },
+        newTextEdits?: { [key: string]: TextEdit[] }
+    ): void {
+        if (!newTextEdits) { return; }
+
+        Object.entries(newTextEdits).forEach(([key, edits]) => {
+            if (!allTextEdits[key]) {
+                allTextEdits[key] = [];
+            }
+            allTextEdits[key].push(...edits);
+        });
+    }
+
+    private optimizeTextEdits(allTextEdits: { [key: string]: TextEdit[] }): { [key: string]: TextEdit[] } {
+        const optimizedEdits: { [key: string]: TextEdit[] } = {};
+
+        Object.entries(allTextEdits).forEach(([key, edits]) => {
+            if (edits.length === 0) { return; }
+
+            const sortedEdits = this.sortTextEdits(edits);
+            const combinedEdit = this.combineTextEdits(sortedEdits);
+
+            optimizedEdits[key] = [combinedEdit];
+        });
+
+        return optimizedEdits;
+    }
+
+    private sortTextEdits(edits: TextEdit[]): TextEdit[] {
+        return edits.sort((a, b) => {
+            if (a.range.start.line !== b.range.start.line) {
+                return a.range.start.line - b.range.start.line;
+            }
+            return a.range.start.character - b.range.start.character;
+        });
+    }
+
+    private combineTextEdits(edits: TextEdit[]): TextEdit {
+        const formattedTexts = edits.map((edit, index) => {
+            const text = edit.newText.trim();
+            return index < edits.length - 1 ? `${text},` : text;
+        });
+
+        return {
+            range: edits[0].range,
+            newText: formattedTexts.join('\n').trimStart()
+        };
+    }
+
+    private async updateSourceCode(params: { textEdits: { [key: string]: TextEdit[] } }): Promise<void> {
+        try {
+            await updateSourceCode(params);
+        } catch (error) {
+            console.error("Failed to update source code:", error);
+            throw new Error("Source code update failed");
+        }
+    }
+
+    private async updateInlineDataMapperView(params: InlineAllDataMapperSourceRequest): Promise<void> {
+        try {
+            const finalCodedataResp = await StateMachine
+                .langClient()
+                .getDataMapperCodedata({
+                    filePath: params.filePath,
+                    codedata: params.codedata,
+                    name: params.varName
+                });
+
+            updateInlineDataMapperView(finalCodedataResp.codedata);
+        } catch (error) {
+            console.error("Failed to update inline data mapper view:", error);
+            throw new Error("View update failed");
+        }
+    }
```

Comment:
Let's move these supporting functions to the `utils.ts` file to keep this file focused solely on RPC-related logic and maintain better separation of concerns.

https://github.com/wso2/vscode-extensions/blob/87c9f97b888b047f9d8fd88f50f37f644de35d62/workspaces/ballerina/ballerina-extension/src/rpc-managers/inline-data-mapper/utils.ts